rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1064,11 +1137,10 @@ module RustDataFlow implements InputSig<Location> {
         pat = node1.asPat() and
         (
           // Pattern destructs a struct-like variant.
-          recordVariantDestruction(pat.getPat(),
-            c.(VariantFieldContent).getVariantCanonicalPath(field))
+          recordVariantDestruction(pat.getPat(), c.(VariantRecordFieldContent).getVariant(field))
           or
           // Pattern destructs a struct.
-          structDestruction(pat.getPat(), c.(StructFieldContent).getStructCanonicalPath(field))
+          structDestruction(pat.getPat(), c.(StructFieldContent).getStruct(field))
```

Comment:
I find this `getStruct` (and `getVariant`) a bit hard to read. It looks like it gets a struct _for_ `field`. The name `hasFieldAndGetStruct` would have been a more clear. But it's the same on `main`, so this is more an idea for a future refactor (that I could do). I think separating this into a `getField\0` and `getSruct\0` would be easier to read (but more verbose).